agent/src/AgentGlobalState.ts

```
@@ -21,8 +23,47 @@ export class AgentGlobalState implements vscode.Memento {
         this.globalStorage.set('completion.inline.hasAcceptedFirstCompletion', true)
         this.globalStorage.set('extension.hasActivatedPreviously', 'true')
     }
+
+    public setPersistencePath(path: string): void {
+        this.path = path
+        const storage = this.readFromDisk(path)
+        for (const [key, value] of storage ?? []) {
+            if (this.globalStorage.has(key) && this.globalStorage.get(key) !== value) {
+                throw new Error(
+                    `Global state key ${key} already exists with a different value: (local) ${this.globalStorage.get(
+                        key
+                    )}, (disk) ${value}`
+                )
+            }
+            this.globalStorage.set(key, value)
+        }
+    }
+
+    // Write the contents of the globalStorage to the file at `this.path`
+    // in a format that can be deserialized later back into a Map.
+    private syncToDisk(): void {
+        if (this.path) {
+            const json = JSON.stringify([...this.globalStorage])
+            fs.writeFileSync(this.path, json)
+        }
+    }
+
+    // deserialize the contents of the file at `this.path` into the globalStorage
+    private readFromDisk(path: string): Map<string, any> | undefined {
```

Let's `logError(...)` when this fails. The user may have manually edited the JSON file and we don't want to silently ignore it.